### PR TITLE
Infrastructure: add Ava ESLint plug-in with recommended rules

### PR DESCRIPTION
**Pull request by nschonni**
- No strict declaration in Ava module files
- Ignore the "index" test file as it isn't directly called
- Fix async/await functions
- `after.always` hook must come before `beforeEach`

### CODE REVIEW

**Comment by nschonni** [comment]
@spectranaut don't think this was catching anything really functional except maybe the hook ordering, but I think you know this stuff better

**Comment by nschonni** [comment]
Rebased and ran the cleanup for the new Prettier formatting

File: test/.eslintrc.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "extends": "../.eslintrc.json",
+  "extends": ["../.eslintrc.json", "plugin:ava/recommended"],
```
Comment (spectranaut) [inline test/.eslintrc.json:2]: cool!

**Review by spectranaut — APPROVED** [review]
Looks great! Thanks! :)

**Comment by nschonni** [comment]
Looks like the new regression test comments are working!

**Comment by mcking65** [comment]
curiosity ... @nschonni  @spectranaut 
Why remove use strict from the test module files?

**Comment by nschonni** [comment]
@mcking65 I'd have to go back to the error message, but I think it's related to how Ava loads the tests as modules making it unnecessary

**Pull request merged**